common/socket/src/main/java/io/helidon/common/socket/PlainSocket.java

```diff
@@ -38,20 +39,23 @@ public sealed class PlainSocket implements HelidonSocket permits TlsSocket {
     private final Socket delegate;
     private final String childSocketId;
     private final String socketId;
+    private final RequestedUriDiscoveryContext discoveryContext;
     private final InputStream inputStream;
     private final OutputStream outputStream;
 
     /**
-     * Plain socket.
+     * Creates a new {@code PlainSocket}.
      *
      * @param delegate delegate socket
      * @param childSocketId channel id
      * @param socketId  server channel id
+     * @param discoveryContext requested URI discovery context
      */
-    protected PlainSocket(Socket delegate, String childSocketId, String socketId) {
+    protected PlainSocket(Socket delegate, String childSocketId, String socketId, RequestedUriDiscoveryContext discoveryContext) {
```

Comment:
I think that socket is not the right point to store the discovery context. This belongs to `ListenerContext` (there is a PR adding it in progress), or to `ConnectionContext` which we use right now (before my PR)

Reply:
Fixed. Added to listener instead.